### PR TITLE
Add pipelines to run existing cpu pytests on AzureML and update yaml pipelines

### PR DESCRIPTION
**Pull request by bethz**
### Description
- Add ability to run cpu based pytests on AzureML. Note: GPU tests are not included in this PR due to issues in the repo.
- removed ci trigger from yaml pipeline files as needed

### Checklist:

- [x] My code follows the code style of this project, as detailed in our [contribution guidelines](../CONTRIBUTING.md).
- [x] I have added tests.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: tests/ci/aml_cpu_unit_tests.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+      - script: pip install paramiko==2.4.2 &&
+          pip install azureml-sdk &&
+          pip install azure-cli
+        displayName: "install azureml-sdk"
+
+      - script: python tools/generate_conda_file.py --name nlp
+        displayName: " generate_conda_file.py"
+
+      - script: python tests/ci/submit_azureml_pytest.py --subid $(SubscriptionID) --testfolder $(testfolder) --testmarkers $(testmarkers) --clustername $(clustername) --expname $(expname) --dockerproc $(dockerproc) --junitxml $(junitxml) --reponame $(reponame) --branch $(Build.SourceBranchName)
```
Comment (miguelgfierro) [inline tests/ci/aml_cpu_unit_tests.yml]: could you please add here `--condafile` explicitely? When seeing this it was confusing to me that the conda file is being generated and used in the submit_azureml_pytest.py with its default value

File: tests/ci/aml_cpu_unit_tests.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    value: "./tests/unit"
+  - name: "testmarkers"
+    value: '"not notebooks and not gpu and not azureml"'
+  - name: "junitxml"
+    value: "reports/test-unit.xml"
+  - name: "maxnodes"
+    value: 4
+  - name: "reponame"
+    value: "nlp"
+  - name: "clustername"
```
Comment (miguelgfierro) [inline tests/ci/aml_cpu_unit_tests.yml]: We could review these parameters, maybe they are creating noise. If the parameter is used several times, then it makes sense to have a variable. However, if the parameter is used once, we will have:
```
  - name: "junitxml"
    value: "reports/test-unit.xml"

python tests/ci/submit_azureml_pytest.py [...] --junitxml $(junitxml)
```
That for every parameter. However, if we remove the non-repeated variables there will be only one line:
```
python tests/ci/submit_azureml_pytest.py [...] --junitxml "reports/test-unit.xml
```
I think the second option helps with readability. 

I'm looking at [other pipelines](https://github.com/microsoft/LightGBM/blob/master/.vsts-ci.yml) and I don't see that many options.

Do people prefer to have everything parametrized? @yueguoguo @saidbleik @jiata

**Comment by saidbleik** [comment]
@miguelgfierro do you know how the variable group can be authorized?

File: README.md [hunk 3]
```diff
@@ -65,3 +65,8 @@ The following is a list of related repositories that we like and think are usefu
 | --- | --- | --- | --- | --- | --- |
 | **Linux CPU** | master | [![Build Status](https://dev.azure.com/best-practices/nlp/_apis/build/status/cpu_integration_tests_linux?branchName=master)](https://dev.azure.com/best-practices/nlp/_build/latest?definitionId=50&branchName=master) | | staging | [![Build Status](https://dev.azure.com/best-practices/nlp/_apis/build/status/cpu_integration_tests_linux?branchName=staging)](https://dev.azure.com/best-practices/nlp/_build/latest?definitionId=50&branchName=staging) |
 | **Linux GPU** | master | [![Build Status](https://dev.azure.com/best-practices/nlp/_apis/build/status/gpu_integration_tests_linux?branchName=master)](https://dev.azure.com/best-practices/nlp/_build/latest?definitionId=51&branchName=master) | | staging | [![Build Status](https://dev.azure.com/best-practices/nlp/_apis/build/status/gpu_integration_tests_linux?branchName=staging)](https://dev.azure.com/best-practices/nlp/_build/latest?definitionId=51&branchName=staging) |
+
+## Build Status on AzureML
+| Build Type | Branch | Status |  | Branch | Status |
```
[the remark this answers is not in the document]
Comment (yueguoguo) [reply README.md]: +1 on this - the benefits of using AML should be highlighted once it is done

* Allow non-MS contributor to trigger tests when submitting PR
* Elasticity in terms of selecting testing resources (e.g., CPU cluster, GPU, and/or Spark cluster) for maximizing testing efficiency
* Configurability of testing environment by using pre-built Docker image

**Comment by yueguoguo** [comment]
@saidbleik the variable group may require keyvault and this is yet to be created AFAIK

**Comment by sharatsc** [comment]
Needs to be resent with the new code changes in place.

**Pull request closed without merging**